MainDemo.Wpf/Domain/PasswordHelper.cs

```diff
@@ -0,0 +1,69 @@
+﻿namespace MaterialDesignDemo.Domain;
+
+/// <summary>
+/// Simple helper class to enable data-binding for <see cref="PasswordBox.Password"/> which is not a <see cref="DependencyProperty"/>
+/// </summary>
+internal static class PasswordHelper
+{
+    public static readonly DependencyProperty PasswordProperty =
+        DependencyProperty.RegisterAttached("Password",
+        typeof(string), typeof(PasswordHelper),
+        new FrameworkPropertyMetadata(string.Empty, OnPasswordPropertyChanged));
+
+    public static readonly DependencyProperty AttachProperty =
```

Comment:
I think if we create a behavior for doing the syncing (like what is in #2812), I suspect we will not need this property as the registration will be done in a behavior, so attaching the behavior to the PasswordBox is the "trigger" for registering for the needed events.

Reply:
I agree that a behavior is a better approach. However, a behavior is typically a dedicated type. I still think we should create a `PasswordBoxAssist` type which can hold attached properties for the style and such, but I don't think that class itself should be the behavior. What do you think is the best approach? Should we create another type alongside of it (e.g. `PasswordBoxBindingBehavior` or similar)? Or perhaps create it as a nested class under the `PasswordBoxAssist` (e.g. `PasswordBoxAssist.BindingBehavior`)?

Reply:
I am good with that.